anitya/sar.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions
+# of the GNU General Public License v.2, or (at your option) any later
+# version.  This program is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.  You
+# should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+#
+"""
+This script is used for GDPR SAR (General Data Protection Regulation
+Subject Access Requests).
+It returns information about specific user saved by Anitya.
+It reads SAR_USERNAME and SAR_EMAIL environment variables as keys for
+getting data from database.
+
+ Authors:
+   Michal Konecny <mkonecny@redhat.com>
+"""
+
+import logging
+import os
+import json
+import sys
+
+from anitya.config import config
+from anitya import db
+
+_log = logging.getLogger('anitya')
+
+SAR_USERNAME = os.getenv('SAR_USERNAME')
+SAR_EMAIL = os.getenv('SAR_EMAIL')
+
+
+def main():
+    '''
+    Retrieve database entry for user.
+    '''
+    db.initialize(config)
+    session = db.Session()
+    _log.setLevel(logging.DEBUG)
+
+    users = []
+
+    if SAR_EMAIL:
+        _log.debug('Find users by e-mail {}'.format(SAR_EMAIL))
+        users = users + session.query(db.User).filter(db.User.email == SAR_EMAIL).all()
```

Comment:
You can write this as ``db.User.query..filter(db.User.email == SAR_EMAIL).all()`` or even ``db.User.query.filter_by(email=SAR_EMAIL).all()``.

Reply:
Didn't know you can write it like this. It is more readable. I will change it.